ts/src/renderer/phaser/scenes/GameScene.ts

```diff
@@ -189,15 +191,55 @@ class GameScene extends PhaserScene {
 				map.addTilesetImage(tileset.name, key);
 			}
 		});
+
+		// instantiate the array that will hold the Layer references
+		this.layers = [];
+
 		data.map.layers.forEach((layer) => {
+
+			// floor, 0
+			// floor2, 1
+			// debris, 2 (returns early)
+			// walls, 3
+			// trees, 4
+
 			if (layer.type !== 'tilelayer') {
 				return;
 			}
-			console.log(layer.name);
-			const tilemapLayer = map.createLayer(layer.name, map.tilesets, 0, 0);
-			tilemapLayer.setScale(scaleFactor.x, scaleFactor.y);
+
+			map.createLayer(layer.name, map.tilesets, 0, 0)
+				.setScale(scaleFactor.x, scaleFactor.y)
+				.setName(`map: ${layer.name}`);
+
+			// add to array this.layers
+			this.layers.push(
+				// create Layer and add to scene
+				this.add.layer()
+					// give it a name so we can track it
+					.setName(layer.name)
+			);
 		});
 
+		// since we returned early for 'debris', it needs a Layer still.
+		// it goes at the index after 'walls'
+		this.layers.splice(
+			3,
+			0,
+			this.add.layer().setName('debris')
+		);
+
+		// 'debris' Layer needs to be inserted at the correct index in the Scene's displayList
+		// it goes right above the Layer for 'walls'.
+		// this logic is for backwards compatibility with taro
+		const walls = this.children.getByName('walls');
+		const debris = this.children.getByName('debris');
+
+		this.children.moveTo(
+			debris,
+			// index of walls layer + 1
```

Comment:
this can be omitted too, seems very clear what the line below does.
comments should be added for giving context or clarifying non-obvious things, rather than putting in words what self-explanatory code does